### ISSUE TITLE
Memory leak in TreeList

### ISSUE THREAD

**Issue by mbechev** [telerik/kendo-angular#3746]
**Describe the bug**
When the TreeList component is rerendered multiple times, a memory leak is observed. Here is a stats comparison between the Grid and TreeList:

### Grid:
![image](https://user-images.githubusercontent.com/33104481/185634934-66080baa-f784-4c87-a05f-904ea6b296d5.png)

### TreeList:
![image](https://user-images.githubusercontent.com/33104481/185634957-7dac8ce2-63e9-466e-a17c-ff6afbda845b.png)

**To Reproduce**
[treelist-A14.zip](https://github.com/telerik/kendo-angular/files/9382707/treelist-A14.zip)

1. Download and unzip the folder.
2. Run **npm install**.
3. Execute **ng serve**.

**Comment by mbechev** [comment]
Consider fixing this issue in the previous major **v2.x.x** of the component.

**Comment by mbechev** [comment]
A client observed that the memory leak is probably caused by the subscription in the **BrowserSupportService** class. 

The implements `ngOnDestroy` where the subscription is unsubscribed:

```ts
@Injectable()
export class BrowserSupportService implements OnDestroy {
    constructor(private zone: NgZone, private changeDetector: ChangeDetectorRef) {
        ...
        this.zone.runOutsideAngular(() => {
            this.subscriptions = fromEvent(window, 'resize').pipe(
                auditTime(100)
            ).subscribe(() => {
                if (cachedPixelRatio !== window.devicePixelRatio) {
                    ...
                }
            });
        });
    }

    public ngOnDestroy(): void {
        if (this.subscriptions) {
            this.subscriptions.unsubscribe();
            this.subscriptions = null;
        }
    }
```

But the TreeList doesn't implement `OnDestroy` hook at all, hence the subscription is not unsubscribed.

**Comment by mbechev** [comment]
Fixed in [`v13.1.0`](https://www.telerik.com/kendo-angular-ui/components/changelogs/kendo-angular-ui/#v13.1.0).
